### PR TITLE
getDataString returns null, so get tag data from EXTRA_NDEF_MESSAGES

### PR DESCRIPTION
**Pull request by sparkhead95**
Intent.getDataString() returns null so the user cannot see tag contents.
Instead get the tag data as a parcelable extra, and parse it that way. I've tested this on API 23 - 25 (works on both).
The min SDK can be reduced to 23 too.

### CODE REVIEW

File: .idea/modules.xml [hunk 1]
```diff
@@ -3,6 +3,8 @@
   <component name="ProjectModuleManager">
     <modules>
       <module fileurl="file://$PROJECT_DIR$/app/app.iml" filepath="$PROJECT_DIR$/app/app.iml" />
+      <module fileurl="file://$PROJECT_DIR$/app/app.iml" filepath="$PROJECT_DIR$/app/app.iml" />
+      <module fileurl="file://$PROJECT_DIR$/nfc-explorer.iml" filepath="$PROJECT_DIR$/nfc-explorer.iml" />
```
Comment (infertux) [inline .idea/modules.xml]: Why did you duplicate these lines?

File: .idea/misc.xml [hunk 2]
```diff
@@ -29,7 +29,7 @@
       </value>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (infertux) [inline .idea/misc.xml]: Could you revert this or set `project-jdk-name` to `"1.7"` as well for consistency?

**Review by infertux — CHANGES_REQUESTED** [review]
Looks good except for some XML quirks.

File: .idea/modules.xml (hunk 1 above)
Comment (sparkhead95) [reply .idea/modules.xml]: Oops, my IDE did this, sorry thought I discarded the change

File: .idea/misc.xml (hunk 2 above)
Comment (sparkhead95) [reply .idea/misc.xml]: Sure, IDE taking over again!

**Comment by infertux** [comment]
@sparkhead95 Good stuff, thanks!

**Pull request merged**